ragaai_catalyst/ragaai_catalyst.py

```
@@ -165,6 +170,29 @@ def _refresh_token_async(self):
+        if not self._token_expiry:
+            return
+            
+        # Calculate when to refresh (20 seconds before expiration)
+        current_time = time.time()
+        refresh_buffer = min(20, RagaAICatalyst.TOKEN_EXPIRY_TIME * 0.05)  # 20 seconds or 5% of expiry time, whichever is smaller
+        time_until_refresh = max(self._token_expiry - current_time - refresh_buffer, 1)  # At least 1 second
+        
+        def delayed_refresh():
+            # Sleep until it's time to refresh
+            time.sleep(time_until_refresh)
+            logger.debug(f"Scheduled token refresh triggered")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)